### PR TITLE
Standalone model and evaluate on root files

### PR DESCRIPTION
**Pull request by badeaa3**
Major change: 
- `spanet/network/jet_reconstruction/jet_reconstruction_network.py` is now a standalone model which includes the training/validation steps and can work in evaluation mode. I did this by copying over the relevant code from `jet_reconstruction_base.py`, `jet_reconstruction_training.py`, and `jet_reconstruction_validation.py`. Most importantly for this PR, this removes the need to supply an input file to construct the network and allows evaluation directly on data using `model.predict_jets_and_particles(source_data, source_mask)`, which enables the user to load data from a root. 
- `spanet/options.py` default optimizer AdamW
- `evaluate.py` standalone evaluation script which can evaluate in parallel 
- `train_test.py` **proposed** training script to overwrite use instead of existing one. This script is heavily based on the existing one. 
- `event_files/GG_rpv_viaN1.ini` event file for 2x5 model (successfully training). Note that the number of children per gluino is inferred in `jet_reconstruction_network.py` from the event_file and currently only supports symmetric pair production (i.e. assumes that num_children is the same for the resonances). For more complicated decays we will need to update that one line of logic.

Before merging I would like @jbossios to review and provide comments based on his experience. I'd like to rename `train_test.py` to `train.py` and archive old training script as well.  

I'd also like to consider if we should copy over the relevant code into the ATLAS analysis gitlab?

### CODE REVIEW

**Comment by badeaa3** [comment]
One thing I'm a bit worried by is that training the 2x5 model with only a batch size of 500 takes 7GB of ram on GPU.

**Comment by jbossios** [comment]
Suggestion: save minJetPt and maxNjets settings when training (those used when creating the input H5 file), that way, when evaluating we can read that information from the spanet output and coherently evaluate on root files using same cuts as used when creating the H5 inputs to training.
